VenueControl/VenueAuthoring/PropertyEntrySessionStates/ScheduleEntry/ConsistentOpeningTimeEntryState.cs

```diff
@@ -68,9 +70,20 @@ public Task OnMessageReceived(MessageVeniInteractionContext c)
                 return c.Session.MoveStateAsync<ConsistentOpeningTimeEntrySessionState>(c);
             }
 
-            // setting closing times
+            if ( ! authorizer.Authorize(c.Interaction.Author.Id, Permission.SetLongSchedule, this._venue).Authorized)
+            {
+                var start = new TimeOnly(_venue.Schedule[0].Start.Hour, _venue.Schedule[0].Start.Minute);
+                var end = new TimeOnly(hour, minute);
+                var diff = end - start;
+                if (diff > TimeSpan.FromHours(7)) 
+                    return c.Interaction.Channel.SendMessageAsync(VenueControlStrings.OpeningTooLong);
+            }
```

Comment:
If user doesn't have the new `SetLongSchedule` permission then we now limit them to 7 hours. I checked the statistics on average opening times for this choice, so I'm confident only a small percentage will see this, and these would be the one we want to check with. 
Engineer not required to override it, I'll add the permission to the Indexers when we go live.